### PR TITLE
Adds new annotation to provide return values to future returning methods

### PR DESCRIPTION
**Pull request by robert-northmind**
## Description
Adds a new annotations for handling `Future` return values.
And also refactors existing annotations and renames them to more descriptive names.
This is an API breaking change!
Solves issue #74 

**Background**
When your driver has a method which returns a future. Then you currently run into problem.
Since if you want to provide a default value using @DriverAction then the value which you pass in there, it needs to be a const value. Since the annotation needs to have a const constructor.

And you cannot out-of-the-box create a const-future.

**Solution**
We add a new annotation which you can use when your function is returning a future.
The new annotation is called: `TestDriverDefaultFutureValue` and into it you pass the return value of the future which you are returning. See this example for how you can use this:

```
@GenerateTestDriver
class CoffeeLibraryPageDriver extends WidgetDriver {
  @TestDriverDefaultFutureValue(false)
  Future<bool> getCoffees() async {
    ... // Some logic..
    return false;
  }
}
```

That would now generate this code:
```
class _$TestCoffeeLibraryPageDriver extends TestDriver
    implements CoffeeLibraryPageDriver {
  @override
  Future<bool> getCoffees() {
    return Future.value(false);
  }
}
```

## More info

This PR also removes the old existing annotations and replaces them with more descriptive annotations.
The new annotation used for the driver class to make sure it gets generated it `GenerateTestDriver`.
And then the two annotations for properties and actions are merged into one which is called: `TestDriverDefaultValue`

## Even More info
This is a multi step rocket.

1. First we update the annotations to have this new annotation.
2. Then we update the generator to generate that new code using this new annotation
3. Update the example app to showcase how to use this new annotation

## Type of Change

- [ ] 🚀 New feature (non-breaking change)
- [ ] 🛠️ Bug fix (non-breaking change)
- [x] ⚠️ Breaking change (feature or bug fix which breaks existing behaviors/APIs)
- [x] 🏗️ Code refactor
- [ ] ⚙️ Build configuration change
- [x] 📝 Documentation
- [ ] 🧹 Chore / Housekeeping

### CODE REVIEW

**Comment by robert-northmind** [comment]
This is still a draft PR.. I will add some unit tests here also.
They will come tomorrow

File: widget_driver_annotation/pubspec.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: widget_driver_annotation
 description: Defines the annotations used by widget_driver and widget_driver_generator to create code for your WidgetDrivers
-version: 0.0.2
+version: 0.0.3
```
Comment (robert-northmind) [inline widget_driver_annotation/pubspec.yaml:3]: Is this okay? Or should I bump to `0.1.0` or `1.0.0` since this is a breaking change?

Comment (JanMichaelPeter) [reply widget_driver_annotation/pubspec.yaml:3]: I think we can stick with this since we assume low usage of this at the moment.

**Review by JanMichaelPeter — DISMISSED** [review]

File: widget_driver_annotation/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
 # CHANGELOG
 
+## 0.0.3
```
[the remark this answers is not in the document]
Comment (robert-northmind) [reply widget_driver_annotation/CHANGELOG.md:3]: haha. it went missing :) 
We forgot to add a changelog message for it..
Should I add something for it also?

[the remark this answers is not in the document]
Comment (robert-northmind) [reply widget_driver_annotation/CHANGELOG.md:3]: I added it now

**Review by ebajrami — APPROVED** [review]

File: scripts/run-tests.sh [hunk 3]
```diff
@@ -10,7 +10,12 @@ PROJECT_FOLDERS=("widget_driver" "widget_driver_annotation" "widget_driver_gener
 function run_tests() {
     current_dir=$(pwd)
     cd $1
-    flutter test --coverage --no-pub --no-test-assets --no-track-widget-creation --no-sound-null-safety || exit $?
+    if [ $project_folder == "widget_driver_annotation" ]; then
+        # The widget_driver_annotation is a dart package, not flutter.
+        dart test --coverage coverage || exit $?
```
Comment (robert-northmind) [inline scripts/run-tests.sh:15]: Can we discuss this tomorrow.
So my test in the annotations package need `dart:mirror`
This package is not available in flutter. only in dart.
So running flutter tests fail. But running dart tests work.

Can I make this change? 
Or should I kill some of my unit tests in annotations?

File: widget_driver_annotation/CHANGELOG.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (4ndrey) [reply widget_driver_annotation/CHANGELOG.md:3]: sorry, it was me, who forgot to update changelogs for all packages 😐

**Review by 4ndrey — APPROVED** [review]

**Review by JanMichaelPeter — APPROVED** [review]

**Pull request merged**